src/libraries/System.Text.RegularExpressions/tests/FunctionalTests/RegexIgnoreCaseTests.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Tests;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.RegularExpressions.Tests
+{
+    // This test takes a long time to run since it needs to compute all possible lowercase mappings across
```

Comment:
In my machine, running **just** this test in the max 12 threads takes around 95 seconds. In the end the theory ends up running 16,344 different inputs, and for each of those inputs it is running the Regex.IsMatch twice (in order to confirm that we properly match both ways of the mapping)

This means that if we were to enable the test to run innerloop it would take the overall run from ~40s in my machine to ~2 minutes or so, which is why I have marked it as outerloop. I also didn't add additional test cases for special cases since we already had a lot of coverage for IgnoreCase for all of the i's in all of the cultures that we care about, and we also have good coverage for IgnoreCase in general.

Reply:
Ok. Outerloop is fine, but that's _a lot_ of test cases. Can we just do it as a loop inside a Fact rather than as a Theory with one char per input? Or as a Theory input per culture but not per char? or something like that?

Reply:
Will do.